src/material/schematics/ng-generate/mdc-migration/rules/components/slider/slider-template.ts

```diff
@@ -37,17 +37,55 @@ export class SliderTemplateMigrator extends TemplateMigrator {
 
     visitElements(ast.nodes, (node: compiler.TmplAstElement) => {
       if (node.name === 'mat-slider') {
-        this._getBindings(node);
+        const originalHtml = node.sourceSpan.start.file.content;
+        const bindings = this._getBindings(node);
+        const inputBindings: string[] = [];
+
+        for (let i = 0; i < bindings.length; i++) {
+          const binding = bindings[i];
+
+          if (binding.name === 'value') {
+            // Move the binding to the <input>.
+            const sourceSpan = binding.node.sourceSpan;
+            inputBindings.push(originalHtml.slice(sourceSpan.start.offset, sourceSpan.end.offset));
+            updates.push(this._removeBinding(originalHtml, binding.node));
+          }
+
+          // TODO(wagnermaciel): Finish the remapping of other bindings.
+        }
+
+        const matSliderThumb = inputBindings.length
+          ? `<input matSliderThumb ${inputBindings.join(' ')} />`
+          : '<input matSliderThumb />';
 
         updates.push({
-          offset: node.sourceSpan.start.offset,
-          updateFn: (html: string) => html,
+          offset: node.startSourceSpan.end.offset,
+          updateFn: (html: string) =>
+            html.slice(0, node.startSourceSpan.end.offset) +
+            matSliderThumb +
+            html.slice(node.startSourceSpan.end.offset),
         });
       }
     });
     return updates;
   }
 
+  /** Returns an update that removes the given binding from the given template ast element. */
+  private _removeBinding(originalHtml: string, binding: compiler.TmplAstNode): Update {
+    let charIndex = binding.sourceSpan.start.offset - 1;
+
+    // Find the first char before the binding that is not whitespace.
+    while (/\s/.test(originalHtml.charAt(charIndex))) {
```

Comment:
nit: could this go into an infinite loop if there is no whitespace character? To be on the safe side you could add `&& charIndex > -1` to the `while`.

Reply:
Technically no because if we're starting at a binding it's guaranteed to eventually reach the start of the tag, but this also makes me nervous so I'll add the additional check just in case haha